engine/src/main/java/org/hibernate/search/engine/backend/orchestration/spi/BatchingExecutor.java

```diff
@@ -127,7 +127,7 @@ public void submit(W work, OperationSubmitter operationSubmitter) throws Interru
 					"Attempt to submit a work to executor '" + name + "', which is stopped."
 			);
 		}
-		operationSubmitter.submitToQueue( workQueue, work, blockingRetryProducer );
+		operationSubmitter.submitToQueue( workQueue, work, blockingRetryProducer, work::markAsFailed );
```

Comment:
Use `BatchedWork::markAsFailed` to avoid capturing lambdas, which have a performance impact.
Also, you could put `BatchedWork::markAsFailed` into a `static final` constant.